pkg/policy/api/rule_validation.go

```diff
@@ -150,12 +150,24 @@ func (e *EgressRule) sanitize() error {
 // TODO we need to add support to check
 // wildcard and prefix/suffix later on.
 func (kr *PortRuleKafka) Sanitize() error {
+	if (len(kr.APIKey) > 0) && (len(kr.Role) > 0) {
```

Comment:
I would like to suggest enforcing this in the schema with `OneOf` but given the recent removal I'm not sure we can still do that. @aanm?

Reply:
We can. What we can't do is having any other fields in the **root** of the CRD validation schema besides `"properties"`, so we can't have `OneOf("spec", "specs")` because that validation would need to be applied at the root of the CRD validation. We can still have something like `OneOf("matchLabels", "matchExpressions")` if this `OneOf` is in the "properties" of a non-root field like "EndpointSelector".